src/ray/raylet/node_manager.cc

```diff
@@ -326,6 +326,30 @@ void NodeManager::Heartbeat() {
     heartbeat_data->add_resource_load_capacity(resource_pair.second);
   }
 
+  std::unordered_set<ObjectID> active_object_ids = worker_pool_.GetActiveObjectIDs();
+  uint64_t max_size = RayConfig::instance().raylet_active_object_ids_size();
+  if (active_object_ids.size() <= max_size) {
+    for (const auto &object_id : active_object_ids) {
+      if (heartbeat_data->active_object_id_size() == max_size) {
+        break;
+      }
+      std::string object_id_bytes = object_id.Binary();
+      heartbeat_data->add_active_object_id(object_id_bytes.data(),
+                                           object_id_bytes.size());
+    }
+  } else {
+    // If there are more than the configured maximum number of object IDs to send per
```

Comment:
Oh interesting, should it also shuffle on core worker side too?

Reply:
The shuffling is just to sample up to the configured limit. I think it's probably fine to send the non-truncated list from worker->raylet to keep the logic in one place but would be fine sampling in both places.